src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Hover/DefaultRazorHoverInfoServiceTest.cs

```diff
@@ -492,11 +510,267 @@ public void GetHoverInfo_TagHelper_Attribute_VSClient_ReturnVSHover()
+            var codeDocument = CreateCodeDocument(output);
+            var csharpSourceText = codeDocument.GetCSharpSourceText();
+            var csharpDocumentUri = new Uri("C:/path/to/file.razor__virtual.g.cs");
+            var serverCapabilities = new ServerCapabilities()
+            {
+                HoverProvider = true
+            };
+
+            await using var csharpServer = await CSharpTestLspServerHelpers.CreateCSharpLspServerAsync(csharpSourceText, csharpDocumentUri, serverCapabilities, razorSpanMappingService: null).ConfigureAwait(false);
+            await csharpServer.OpenDocumentAsync(csharpDocumentUri, csharpSourceText.ToString()).ConfigureAwait(false);
+
+            var razorFilePath = "C:/path/to/file.razor";
+            var documentContextFactory = new TestDocumentContextFactory(razorFilePath, codeDocument, version: 1337);
+            var languageServerFeatureOptions = Mock.Of<LanguageServerFeatureOptions>(options =>
+                options.SupportsFileManipulation == true &&
+                options.SingleServerSupport == true &&
+                options.CSharpVirtualDocumentSuffix == ".g.cs" &&
+                options.HtmlVirtualDocumentSuffix == ".g.html"
+                , MockBehavior.Strict);
+            var languageServer = new HoverLanguageServer(csharpServer, csharpDocumentUri);
+            var documentMappingService = new DefaultRazorDocumentMappingService(languageServerFeatureOptions, documentContextFactory, LoggerFactory);
+            var projectSnapshotManager = Mock.Of<ProjectSnapshotManagerBase>(p => p.Projects == new[] { Mock.Of<ProjectSnapshot>(MockBehavior.Strict) }, MockBehavior.Strict);
+            var lspTagHelperTooltipFactory = new DefaultLSPTagHelperTooltipFactory();
+            var vsLspTagHelperTooltipFactory = new DefaultVSLSPTagHelperTooltipFactory();
+            var hoverInfoService = new DefaultRazorHoverInfoService(TagHelperFactsService, lspTagHelperTooltipFactory, vsLspTagHelperTooltipFactory, HtmlFactsService);
+
+            var endpoint = new RazorHoverEndpoint(
+                documentContextFactory,
+                hoverInfoService,
+                languageServerFeatureOptions,
+                documentMappingService,
+                languageServer,
+                LoggerFactory);
+
+            codeDocument.GetSourceText().GetLineAndOffset(cursorPosition, out var line, out var offset);
+            var request = new VSHoverParamsBridge
+            {
+                TextDocument = new TextDocumentIdentifier
+                {
+                    Uri = new Uri(razorFilePath)
+                },
+                Position = new Position(line, offset)
+            };
+
+            // Act
+            var result = await endpoint.Handle(request, CancellationToken.None);
+
+            // Assert
+            var range = result.Range;
+            var expected = new Range()
+            {
+                Start = new Position(line: 32, character: 8),
```

Comment:
Line 32? Seems too high to be razor, but too low to be C#.. are we mapping correctly?

Reply:
I wasn't sure if it was just because of using the `DefaultRazorDocumentMappingService`. I'll dig in a bit

Reply:
This was a bug, I was doing the mapping backwards. It didn't break anything because range isn't used for hover but... hey! Now it's fixed